Sources/BraveWallet/Extensions/WalletTxServiceExtensions.swift

```diff
@@ -30,4 +30,27 @@ extension BraveWalletTxService {
       }
     )
   }
+  
+  // Fetches all transactions for all given keyrings
+  func allTransactions(
+    for keyrings: [BraveWallet.KeyringInfo]
+  ) async -> [BraveWallet.TransactionInfo] {
```

Comment:
Can we merge this with the above `pendingTransactions` helper function? We could keep `allTransactions` name but pass in an optional transaction status to filter for

Reply:
moved to `TransactionConfirmationStore` as a private extension since that is the class actually uses it. [29dc327](https://github.com/brave/brave-ios/pull/6960/commits/29dc3270f2191a8677bd908927acd01b0fc1f7de)

Reply:
it's ok to keep the `allTransactions` private for now since i dont think there is any other place uses it. can reconsider if we have more cases that need this helper function.

Reply:
Access control shouldn't prevent us from re-using code here. I don't think we should have duplicated code like this, it's an easy fix here rather than updating 2 spots later if APIs change.